runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -240,15 +285,55 @@ void registerTable(@Nonnull String namespace,
         }
     }
 
+    /**
+     * For each protobuf filename -> descriptor map, validate if the previous descriptor matches
+     * If there is a new schema or an update, record the new update in the table.
+     * (Note that oldProto means existing schema previously opened)
+     * WARNING: This method MUST be invoked within a transaction.
+     *
+     * @param protoName - name of the protobuf file
+     * @param newProtoFd - descriptor of the protobuf file that is being inserted.
+     * @param namespace - namespace of the table
+     * @param tableName - tablename without the namespace
+     */
+    private void recordNewSchema(ProtobufFileName protoName,
+                                    CorfuRecord<ProtobufFileDescriptor, TableMetadata> newProtoFd,
+                                    String namespace,
+                                    String tableName) {
+        final CorfuRecord<ProtobufFileDescriptor, TableMetadata> oldProto = this.protobufDescriptorTable.get(protoName);
+        if (oldProto == null) {
+            this.protobufDescriptorTable.put(protoName, newProtoFd);
+            return;
+        }
+        if (!oldProto.getPayload().getFileDescriptor().equals(newProtoFd.getPayload().getFileDescriptor())) {
+            if (!protoName.getFileName().startsWith("google/protobuf")) {
```

Comment:
why are we ignoring the files in the google protobuf library?

Reply:
2 reasons:
1. they are default library files, even if there is a schema change there nothing we can or should do about it.
2. They are very huge with complex nested types, comparing these itself is a CPU intensive process.

Reply:
done. Removal is a whole other conversation. We need to provide future releases with a removeTable() functionality so we can safely delete old stale protobufs associated with a deprecated table.